uniffi/src/ffi/rustfuture.rs

```diff
@@ -0,0 +1,586 @@
+//! pub extern "C" fn _uniffi_hello_drop(
+//!     future: Option<Box<::uniffi::RustFuture<bool>>>,
+//!     call_status: &mut ::uniffi::RustCallStatus
+//! ) {
+//!     ::uniffi::ffi::uniffi_rustfuture_drop(future, call_status)
+//! }
+//! ```
+//!
+//! First off, this _drop function_ is responsible to drop the `RustFuture`. It's
+//! clear by looking at its signature: It receives an `Option<Box<RustFuture<_>>>`,
+//! i.e. it takes ownership of the `RustFuture` _via_ `Box`!
+//!
+//! Similarly to the _poll function_, it forwards everything to
+//! `uniffi_rustfuture_drop`, which is the generic version of the monomorphised _drop
+//! function_.
+//!
+//! ## How `Future` works exactly?
+//!
+//! A [`Future`] in Rust does nothing. When calling an async function, it just
+//! returns a `Future` but nothing has happened yet. To start the computation,
+//! the future must be polled. It returns [`Poll::Ready(r)`][`Poll::Ready`] if
+//! the result is ready, [`Poll::Pending`] otherwise. `Poll::Pending` basically
+//! means:
+//!
+//! > Please, try to poll me later, maybe the result will be ready!
+//!
+//! This model is very different than what other languages do, but it can actually
+//! be mangled quite easily, hopefully for us!
+//!
+//! But… wait a minute… who is responsible to poll the `Future` if a `Future` does
+//! nothing? Well, it's _the executor_ (sometimes also called _the runtime_ but not
+//! here). The executor is responsible _to drive_ the `Future`: that's where
+//! they are polled.
+//!
+//! But… wait another minute… how the executor knows when to poll a [`Future`]?
+//! Does it poll them randomly in a forever loop? Well, no, actually it depends
+//! of the executor! A well-designed `Future` and executor work as follows.
+//! Normally, when [`Future::poll`] is called, a [`Context`] argument is
+//! passed to it. It contains a [`Waker`]. The [`Waker`] is built on top of a
+//! [`RawWaker`] which implements whatever is necessary. Usually, a waker will
+//! signal the executor to poll a particular `Future`. A `Future` will clone
+//! or pass-by-ref the waker to somewhere, as a callback, a completion, a
+//! function, or anything, to the system that is responsible to notify when a
+//! task is completed. So, to recap, the waker is _not_ responsible to wake the
+//! `Future`, it _is_ responsible to _signal_ the executor that a particular
+//! `Future` must be polled again. That's why the documentation of
+//! [`Poll::Pending`] specifies:
+//!
+//! > When a function returns `Pending`, the function must also ensure that the
+//! > current task is scheduled to be awoken when progress can be made.
+//!
+//! “awoken” is done by using the `Waker`.
+//!
+//! ## Awaken from the foreign language land
+//!
+//! Our _poll function_ receives a waker function pointer, along with a waker
+//! environment. We said that the waker function lives in the foreign language
+//! land. That's really important. It cannot live inside Rust because Rust
```

Comment:
```suggestion
//! environment. We said that the waker function lives on the foreign language
//! side. That's really important. It cannot live inside Rust because Rust
```